core/unit_test/TestGraph.hpp

```diff
@@ -208,9 +223,25 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), DISABLED_repeat_chain) {
 }
 
 TEST_F(TEST_CATEGORY_FIXTURE(graph), zero_work_reduce) {
-  auto graph = Kokkos::Experimental::create_graph(ex, [&](auto root) {
-    root.then_parallel_reduce(0, set_result_functor{bugs}, count);
-  });
+  auto graph = Kokkos::Experimental::create_graph(
+      ex, [&](Kokkos::Experimental::GraphNodeRef<TEST_EXECSPACE> root) {
+        NoOpReduceFunctor<TEST_EXECSPACE, int> no_op_functor;
+        root.then_parallel_reduce(Kokkos::RangePolicy<TEST_EXECSPACE>(0, 0),
+                                  no_op_functor, count)
+#if !defined(KOKKOS_ENABLE_SYCL)  // FIXME_SYCL
+#if !defined(KOKKOS_ENABLE_CUDA) && \
+    !defined(KOKKOS_ENABLE_HIP)  // FIXME_CUDA FIXME_HIP
+            .then_parallel_reduce(
+                Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<2>>{{0, 0},
+                                                                       {0, 0}},
+                no_op_functor, count)
```

Comment:
Why does this one not work with HIP nor CUDA?
What about SYCL?  Shouldn't it work if it is using the default graph implementation?

Reply:
For SYCL in the default implementation, it's the same problem that already required skipping the `launch_six` test (and the problem wasn't obvious to me).
For Cuda and HIP, we don't treat empty kernel launches correctly, i.e., they are not inserted into the graph.

Reply:
Why is it incorrect to skip inserting them if they do nothing?

Reply:
Because we are asserting that the node is not empty after setting it up and we actually need to reset the result variable (which we do correctly for the other policies).

Reply:
Does the test fail with HIP and CUDA? If it passes, we should change the test to expose the problem

Reply:
It fails with HIP and CUDA if not disabling the MDRangePolicy test. After disabling the MDRangePolicy test it passes.